src/core/map/types.ts

```
@@ -0,0 +1,108 @@
+
+  /**
+   * Attempts to render content using registered providers.
+   * @param mapEvent - The original MapLibre mouse event
+   * @param onClose - Callback to close the popover (for interactive content)
+   * @returns Aggregated content from all providers, or null if no provider can handle the event
+   */
+  renderContent(mapEvent: MapMouseEvent, onClose: () => void): React.ReactNode | null;
+}
+
+// Plugin type - can optionally return a cleanup function for unregistering listeners or freeing resources
+export type MapPlugin<TMap extends IMap = IMap> = (map: TMap) => void | (() => void);
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)